eclair-core/src/main/scala/fr/acinq/eclair/swap/SwapInReceiver.scala

```diff
@@ -0,0 +1,247 @@
+                                     |<-------------------------------|
+     (claim_by_coop) [ClaimSwapCoop] |                                |
+
+                                             "Refund After Csv"
+                           [WaitCsv] |                                |
+                                     |                                |
+       (claim_by_csv) [ClaimSwapCsv] |                                |
+
+  */
+
+  def apply(request: SwapInRequest, nodeParams: NodeParams, paymentInitiator: actor.ActorRef, watcher: ActorRef[ZmqWatcher.Command], register: actor.ActorRef, wallet: OnChainWallet): Behavior[SwapCommand] =
+    Behaviors.setup { context =>
+      Behaviors.receiveMessagePartial {
+        case StartSwapInReceiver() =>
+          ShortChannelId.fromCoordinates(request.scid) match {
+            case Success(shortChannelId) => new SwapInReceiver(request, shortChannelId, nodeParams, paymentInitiator, watcher, register, wallet, context)
+              .CreateSwap()
+            case Failure(e) => context.log.error(s"received swap request with invalid shortChannelId: $request, $e")
+              Behaviors.stopped
+          }
+        case RestoreSwapInReceiver(d) =>
+            new SwapInReceiver(d.request, d.shortChannelId, nodeParams, paymentInitiator, watcher, register, wallet, context)
+                .AwaitOpeningTxConfirmed(d.agreement, d.openingTxBroadcasted)
+        case AbortSwapInReceiver => Behaviors.stopped
+      }
+    }
+}
+
+private class SwapInReceiver(request: SwapInRequest, implicit val shortChannelId: ShortChannelId, implicit val nodeParams: NodeParams, paymentInitiator: actor.ActorRef, implicit val watcher: ActorRef[ZmqWatcher.Command], implicit val register: actor.ActorRef, implicit val wallet: OnChainWallet, implicit val context: ActorContext[SwapCommands.SwapCommand]) {
+  val protocolVersion = 1
+  val noAsset = ""
+  private val openingConfirmedAdapter = context.messageAdapter[ZmqWatcher.WatchTxConfirmedTriggered](OpeningConfirmed)
+  private val claimByInvoiceConfirmedAdapter = context.messageAdapter[ZmqWatcher.WatchTxConfirmedTriggered](ClaimTxConfirmed)
+  implicit val timeout: Timeout = 30 seconds
+
+  private val keyManager: SwapKeyManager = nodeParams.swapKeyManager
+  private val feeRatePerKw: FeeratePerKw = nodeParams.onChainFeeConf.feeEstimator.getFeeratePerKw(target = nodeParams.onChainFeeConf.feeTargets.fundingBlockTarget)
+  private val premium = (feeRatePerKw * claimByInvoiceTxWeight / 1000).toLong // TODO: how should swap receiver calculate an acceptable premium?
+  private val swapId: String = request.swapId
+  private val takerPrivkey: PrivateKey = keyManager.openingPrivateKey(SwapKeyManager.keyPath(swapId)).privateKey
+  private val takerPubkey: PublicKey = takerPrivkey.publicKey
+  private val makerPubkey: PublicKey = PublicKey(ByteVector.fromValidHex(request.pubkey))
+
+  private def CreateSwap(): Behavior[SwapCommand] = {
```

Comment:
nit: methods should start with a lowercase letter (this would be consistent with how we define behaviours in the rest of the code)

Reply:
fixed in [eb3e7d9](https://github.com/ACINQ/eclair/pull/2342/commits/eb3e7d9ca8a3d00a8678cd48f5ef4185aae9aa01)